fiducial_follow/README.md

```diff
@@ -5,25 +5,115 @@
+* `max_linear_rate`: the maximum approach speed used when following in meters per sec. Default `1.0`.
+* `max_angular_rate`: the maximum rotation speed used when following in radians per sec. Default `1.2`.
+* `search_for_target`: rotate to find the fiducial if it is not in view. Default `True`.
+* `lost_angular_rate`: angular rotation rate in radians per sec used in searching. Default `0.6`.
+* `drive_rate`: rate in meters per second for the simple drive commands.  Default `0.4`.
+* `rotate_rate`: rate in radians per second for the simple rotate commands.  Default `1.5`.
 
 ### Publications
 
-`cmd_vel`(geometry_msgs/Twist): commands to move the robot.
+* `cmd_vel` (geometry_msgs/Twist): commands to move the robot.
+* `follower_status` (custom_msgs/FollowerStatus): shows status as commands are received, started, and completed
 
 ### Subscriptions
 
-`fiducial_transforms`:(fiducial_msgs/FiducialTransformArray)
+* `fiducial_transforms` (fiducial_msgs/FiducialTransformArray): Listen for locations of observed fiducials
+* `follower_commands` (custom_msgs/FollowerCommand): Accept commands to process in order of reception
+
+### Example usage for simple single fiducial following
+For the simple legacy fiducial follow mode a single aruco fiducial of size 140mm on edge of the black pattern that is number 49 is used.
+Prepare such a fiducial and place it where the robot can see it with the camera then start the program.
+
+    roslaunch magni_demos fiducial_follow.launch
+
+### Usage for following a sequence of fiducials and/or performing simple movements
+
+The follow.py script can act on commands sent to it using ros topic /follower_commands.
+These commands arrive in a general message, FollowerCommand, upon a well known topic of /follower_commands that is monitored by the follower script. 
+
+A powerful application of the following of a sequence of commands is to have the robot follow fiducials on the floor with a downward facing camera configured for your robot.  For optimal floor fiducial following there are some configuration changes suggested where the config files will be present starting from a path of /opt/ros/kinetic/share/raspicam_node for the case of ROS Kinetic.  Floor fiducials of 140mm size up to almost a 2 meters away have been recognized with success. 
+
+#### Optimal following of fiducials on the floor requires Config Changes
+The camera resolution should be increased to perhaps 1640x1232 so detection of fiducials from over a meter away is possible.  This allows greater spacing of fiducials on the floor.   Also it is best to turn off searching.
+* Camera config file must be formed and present, for this case it would be in camera_info/camerav2_1640x1232.yaml  The config file is best formed from a camera calibration done at the resolution, in this case 1640x1232.
+* A raspicam launch file must be present, for this case in launch/camerav2_1640x1232_10fps.launch
+* The launch file should specify argument 'search_for_target' as  'false'
+
```

Comment:
Could you create a PR to add the launch file to raspicam_node instead of describing how to create it.

Reply:
Yes that should happen but it is a chicken and egg thing.  I would first create the raspicam yaml and launch and THEN get back to this to have a launch of  fiducial_floor_follow.launch.
One blocker on that is if a full on calibration has to happen OR can do like I did today (cheap shot) and form the yaml and raspicam launch for 1640x1232 with copies of the 1280x960 files edited.   Does the plum-bob stuff still hold true if I just scale up the pixel count or is new cal required.   In my own environment now I did the copies then edits to form the required 1640x1232 files and it runs but I bet maybe not a good cal.

Reply:
Yeah, this comes up quite a bit, because we have multiple repos.  What you could do is wait until both PRs are approved and then merge them at the same time.

The camera calibration includes image size, the focal length in both directions, the optical center of the image and the distortion coefficients.  You can change the image size and center for the new size and it will probably work quite well, but it is better to do a calibration with a checkerboard.   I have one you can borrow.

Reply:
It might be worth committing the dubious calibration now and creating an issue to create a better one.  That way it is documented that it needs to be done.